x-pack/plugins/security/server/routes/api_keys/create.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+import { ApiKey } from '../../../common/model';
+import { wrapError, wrapIntoCustomErrorResponse } from '../../errors';
+import { RouteDefinitionParams } from '..';
+
+interface ResponseType {
+  itemsInvalidated: Array<Pick<ApiKey, 'id' | 'name'>>;
+  errors: Array<Pick<ApiKey, 'id' | 'name'> & { error: Error }>;
+}
+
+export function defineCreateApiKeyRoutes({ router, clusterClient }: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/api_key',
+      validate: {
+        body: schema.object({
+          name: schema.string(),
+          expiration: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    createLicensedRouteHandler(async (context, request, response) => {
```

Comment:
The `authc` service already exposes a way to create API Keys. Although the logic is straightforward, we could still benefit from having a single way of doing this:

```diff
diff --git a/x-pack/plugins/security/server/routes/api_keys/create.ts b/x-pack/plugins/security/server/routes/api_keys/create.ts
index dfef44e9b6d..9393df8a5a8 100644
--- a/x-pack/plugins/security/server/routes/api_keys/create.ts
+++ b/x-pack/plugins/security/server/routes/api_keys/create.ts
@@ -7,7 +7,7 @@
 import { schema } from '@kbn/config-schema';
 import { createLicensedRouteHandler } from '../licensed_route_handler';
 import { ApiKey } from '../../../common/model';
-import { wrapError, wrapIntoCustomErrorResponse } from '../../errors';
+import { wrapIntoCustomErrorResponse } from '../../errors';
 import { RouteDefinitionParams } from '..';
 
 interface ResponseType {
@@ -15,7 +15,7 @@ interface ResponseType {
   errors: Array<Pick<ApiKey, 'id' | 'name'> & { error: Error }>;
 }
 
-export function defineCreateApiKeyRoutes({ router, clusterClient }: RouteDefinitionParams) {
+export function defineCreateApiKeyRoutes({ router, authc }: RouteDefinitionParams) {
   router.post(
     {
       path: '/internal/security/api_key',
@@ -27,12 +27,16 @@ export function defineCreateApiKeyRoutes({ router, clusterClient }: RouteDefinit
       },
     },
     createLicensedRouteHandler(async (context, request, response) => {
-      const scopedClusterClient = clusterClient.asScoped(request);
       try {
-        const apiKey = await scopedClusterClient.callAsCurrentUser('shield.createAPIKey', {
-          body: request.body,
+        const apiKey = await authc.createAPIKey(request, {
+          ...request.body,
+          role_descriptors: {},
         });
 
+        if (!apiKey) {
+          return response.badRequest({ body: { message: `API Keys are not available` } });
+        }
+
         return response.ok({ body: apiKey });
       } catch (error) {
         return response.customError(wrapIntoCustomErrorResponse(error));

```